qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -311,65 +311,56 @@ def observable(me_table, init_term=0, mapping="jordan_wigner"):
     return structure.convert_observable(jordan_wigner(mb_obs))
 
 
-def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
-    r"""Reads the Hartree-Fock (HF) electronic structure data file, defines an active
-    space and generates the table of matrix elements required to build the second-quantized
-    spin-projection operator :math:`\hat{S}_z`.
+def spin_z(n_orbitals, mapping="jordan_wigner"):
+    r"""Computes the total spin projection operator :math:`\hat{S}_z` in the Pauli basis.
```

Comment:
Really nice!

Reply:
So just to double check, but `get_spinZ_matrix_elements` never needed to depend on the hf data and molecular structure?

Reply:
The OpenFermion `MolecularData` class contains a lot of information that is picked from the output of HF output of PySCF or Psi4. We don't want, unless it is inevitable, to pass this data structure to our functions constructing general observables because this will constrained this functionalities to deal only with molecules and with this specific data structure. Instead, it is better to pass the concrete information you need to build the observable so any user can come with this info, not necessarily for a molecule, and perform simulations.

Reply:
@josh146 the only information I was using from the molecular data structure was the number of orbitals. But I explained above, this is not convenient. Also, it does not looks nice to pass this monster just to pick an integer number :-).